pkgs/development/tools/misc/gdb/default.nix

```diff
@@ -170,7 +173,11 @@ stdenv.mkDerivation rec {
     ++ lib.optional stdenv.hostPlatform.isMusl "--disable-nls"
     ++ lib.optional stdenv.hostPlatform.isStatic "--disable-inprocess-agent"
     ++ lib.optional enableDebuginfod "--with-debuginfod=yes"
-    ++ lib.optional (!enableSim) "--disable-sim";
+    ++ lib.optional (!enableSim) "--disable-sim"
+    # Workaround for Apple Silicon, "--target" must be "faked", see eg: https://github.com/Homebrew/homebrew-core/pull/209753
+    ++ lib.optional (
+      stdenv.hostPlatform.isDarwin && stdenv.hostPlatform.isAarch64
+    ) "--target=x86_64-apple-darwin";
```

Comment:
Passing a `--target=` flag that hardcodes target name is a bit odd and splits the `--build/--host/--target` manipulation into two places. Would it work if you set `configurePlatforms = [ "target" ]` for `darwin`?

And while at it: is it `aarch64`-specific?

Reply:
GDB don't support `--target=aarch64-apple-darwin`, so we have to target others linux/darwin/none

Reply:
btw, when you put `--target=aarch64-apple-darwin` this derivation will compile fine, but `./bin/gdb` will be missing

Reply:
Oh. I just realized you pass `x86_64` for `aarch64` host /o\